src/formats/notion-api.ts

```diff
@@ -0,0 +1,1210 @@
+ */
+	private async replaceSyncedChildPlaceholders(ctx: ImportContext): Promise<void> {
+		if (this.syncedChildPlaceholders.size === 0) {
+			return;
+		}
+	
+		ctx.status('Replacing synced block child references...');
+	
+		let replacedCount = 0;
+		let filesModified = 0;
+		let importedCount = 0;
+	
+		// Iterate through files that have synced child placeholders (efficient O(1) lookup)
+		for (const [filePath, childIds] of this.syncedChildPlaceholders) {
+			if (ctx.isCancelled()) break;
+		
+			try {
+			// Get the file directly by path (O(1) lookup)
+				const file = this.vault.getAbstractFileByPath(normalizePath(filePath));
+				if (!file || !(file instanceof TFile)) {
+					console.warn(`Could not find synced block file: ${filePath}`);
+					continue;
+				}
+		
+				let content = await this.vault.read(file);
+				const originalContent = content;
+			
+				// Process each child ID that was recorded for this file
+				for (const childId of childIds) {
+				// Try as page first
+					const pageId = childId;
+					const pagePlaceholder = `[[SYNCED_CHILD_PAGE:${pageId}]]`;
+				
+					// Check if this is a page placeholder
+					if (content.includes(pagePlaceholder)) {
+						// Check if page is already imported
+						let pagePath = this.notionIdToPath.get(pageId);
+				
+						if (pagePath) {
+							// Already imported, use existing path
+							const targetFile = this.vault.getAbstractFileByPath(pagePath + '.md');
+							if (targetFile && targetFile instanceof TFile) {
+								const link = this.app.fileManager.generateMarkdownLink(targetFile, file.path);
+								content = content.replace(pagePlaceholder, link);
+								replacedCount++;
+							}
+						}
+						else {
+							// Not imported yet, try to import to Notion Synced Blocks folder
+							try {
+								const syncedBlocksFolder = this.outputRootPath.split('/').slice(0, -1).join('/') + '/Notion Synced Blocks';
+								await this.fetchAndImportPage(ctx, pageId, syncedBlocksFolder);
+								importedCount++;
+						
+								// Now get the path
+								pagePath = this.notionIdToPath.get(pageId);
+								if (pagePath) {
+									const targetFile = this.vault.getAbstractFileByPath(pagePath + '.md');
+									if (targetFile && targetFile instanceof TFile) {
+										const link = this.app.fileManager.generateMarkdownLink(targetFile, file.path);
+										content = content.replace(pagePlaceholder, link);
+										replacedCount++;
+									}
+								}
+							}
+							catch (error) {
+								// Failed to import (no access or error)
+								console.warn(`Failed to import synced child page ${pageId}:`, error);
+								content = content.replace(pagePlaceholder, `**Page** _(no access)_`);
+							}
+						}
+					}
+			
+					// Check if this is a database placeholder
+					const databaseId = childId;
+					const dbPlaceholder = `[[SYNCED_CHILD_DATABASE:${databaseId}]]`;
```

Comment:
Please make a function which takes a `databaseId` and returns this string. It should be used here and in the places where the placeholders are being created to ensure that the format stays in sync. Please do the same for the `SYNCED_CHILD_PAGE` placeholder above too.

Reply:
Done. All placeholder have be handled. https://github.com/obsidianmd/obsidian-importer/pull/444/commits/51a9e94bda34386051c90555f197a9434aa9201a